spec/datadog/core/configuration/option_spec.rb

```diff
@@ -553,81 +553,119 @@
     # | default                    | default             | Option is reset |
     # | rc, default                | rc                  | default         |
     # | rc, default                | default             | rc              |
-    #
-    # Coding in tests is not great, but the alternative is to hardcode 2^n combinations
-    # With n the number of values in the precedence list
-    # (with fleet and customer defined stable config, this will go up to 64 combinations)
-    combinations = (0..Datadog::Core::Configuration::Option::Precedence::LIST.length).flat_map do |n|
-      Datadog::Core::Configuration::Option::Precedence::LIST.combination(n).to_a
-    end
 
-    combinations.each do |precedences|
-      name = if precedences.empty?
-               'empty set'
-             elsif precedences.length == 1
-               precedences.first.name
-             else
-               "combination of #{precedences.map(&:name).join(', ')}"
-             end
-      context "for #{name}" do
-        before do
-          allow(context).to(receive(:instance_exec)) { |value, _| value }
+    # We don't need to test all precedence set combinations.
+    context 'with preset precedence' do
+      before do
+        allow(context).to(receive(:instance_exec)) { |value, _| value }
+      end
+
+      context 'when no precedence value is set and try to unset a precedence that is not set' do
+        subject!(:unset) { option.unset(Datadog::Core::Configuration::Option::Precedence::PROGRAMMATIC) }
```

Comment:
Since it's not used, it should be converted into before block

```suggestion
        before { option.unset(Datadog::Core::Configuration::Option::Precedence::PROGRAMMATIC) }

```

Reply:
Changed in [fae0b1c](https://github.com/DataDog/dd-trace-rb/pull/4610/commits/fae0b1c4d22b89202157ce8439d161da0e9b542e) !